pkg/clean/clean.go

```diff
@@ -33,41 +35,47 @@ var log = logger.Log
 // GetNSXClientFailed  			indicate that could not retrieve nsx client to perform cleanup operation
 // InitCleanupServiceFailed 	indicate that error happened when trying to initialize cleanup service
 // CleanupResourceFailed    	indicate that the cleanup operation failed at some services, the detailed will in the service logs
-func Clean(cf *config.NSXOperatorConfig) error {
+func Clean(ctx context.Context, cf *config.NSXOperatorConfig) error {
 	log.Info("starting NSX cleanup")
 	if err := cf.ValidateConfigFromCmd(); err != nil {
-		return errors.Join(ValidationFailed, err)
+		return errors.Join(nsxutil.ValidationFailed, err)
 	}
 	nsxClient := nsx.GetClient(cf)
 	if nsxClient == nil {
-		return GetNSXClientFailed
+		return nsxutil.GetNSXClientFailed
 	}
 	if cleanupService, err := InitializeCleanupService(cf); err != nil {
-		return errors.Join(InitCleanupServiceFailed, err)
+		return errors.Join(nsxutil.InitCleanupServiceFailed, err)
 	} else if cleanupService.err != nil {
-		return errors.Join(InitCleanupServiceFailed, cleanupService.err)
+		return errors.Join(nsxutil.InitCleanupServiceFailed, cleanupService.err)
```

Comment:
Not your change: Why does `InitializeCleanupService` both return an error, but also have an error attached to `cleanupService` object? This is an anti-pattern - would be better to just return the error in the one spot. Maybe a follow-up change can address this?